odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/repository/reactive/ReactiveCollectorRepositoryImpl.java

```diff
@@ -0,0 +1,124 @@
+
+        return jooqReactiveOperations.mono(query).map(this::mapRecordToDto);
+    }
+
+    @Override
+    public Mono<Page<CollectorDto>> listDto(final int page, final int size, final String nameQuery) {
+        final Select<? extends Record> collectorSelect = paginate(
+            DSL.selectFrom(COLLECTOR).where(listCondition(nameQuery)),
+            COLLECTOR.ID,
+            SortOrder.ASC,
+            (page - 1) * size,
+            size
+        );
+
+        final Table<? extends Record> collectorCTE = collectorSelect.asTable("collector_cte");
+
+        final SelectOnConditionStep<Record> query = DSL.with(collectorCTE.getName())
+            .as(collectorSelect)
+            .select(collectorCTE.fields())
+            .select(NAMESPACE.asterisk())
+            .select(TOKEN.asterisk())
+            .from(collectorCTE.getName())
+            .leftJoin(NAMESPACE).on(NAMESPACE.ID.eq(collectorCTE.field(COLLECTOR.NAMESPACE_ID)))
+            .leftJoin(TOKEN).on(TOKEN.ID.eq(collectorCTE.field(COLLECTOR.TOKEN_ID)));
+
+        return jooqReactiveOperations.flux(query)
+            .collectList()
+            .flatMap(records -> jooqQueryHelper.pageifyResult(
+                records,
+                r -> mapRecordToDto(r, collectorCTE.getName()),
+                fetchCount(nameQuery)
+            ));
+    }
+
+    @Override
+    public Mono<CollectorPojo> getByToken(final String token) {
+        final SelectConditionStep<Record> query = DSL
+            .select(COLLECTOR.asterisk())
+            .from(COLLECTOR)
+            .join(TOKEN).on(TOKEN.ID.eq(COLLECTOR.TOKEN_ID))
+            .where(TOKEN.VALUE.eq(token));
+
+        return jooqReactiveOperations.mono(query).map(r -> r.into(CollectorPojo.class));
+    }
+
+    @Override
+    public Mono<Boolean> existsByNamespace(final long namespaceId) {
+        final Select<? extends Record1<Boolean>> query = jooqQueryHelper.selectExists(
+            DSL.selectFrom(COLLECTOR).where(addSoftDeleteFilter(COLLECTOR.NAMESPACE_ID.eq(namespaceId)))
+        );
+
+        return jooqReactiveOperations.mono(query).map(r -> r.get(0, Boolean.class));
+    }
+
+    private CollectorDto mapRecordToDto(final Record record, final String collectorCteName) {
+        final TokenPojo tokenPojo = jooqRecordHelper.extractRelation(record, TOKEN, TokenPojo.class);
+
+        return new CollectorDto(
+            jooqRecordHelper.remapCte(record, collectorCteName, COLLECTOR).into(CollectorPojo.class),
+            jooqRecordHelper.extractRelation(record, NAMESPACE, NamespacePojo.class),
+            new TokenDto(tokenPojo)
+        );
+    }
+
+    private CollectorDto mapRecordToDto(final Record record) {
+        return new CollectorDto(
+            record.into(COLLECTOR).into(CollectorPojo.class),
+            jooqRecordHelper.extractRelation(record, NAMESPACE, NamespacePojo.class),
+            new TokenDto(jooqRecordHelper.extractRelation(record, TOKEN, TokenPojo.class))
```

Comment:
let's do the same logic as above, otherwise you can have tokendto object with null inside which can lead to NPE in different places